### ISSUE TITLE
database plugin preference cannot be "applied"

### ISSUE THREAD

**Issue by yasenstar** [archi-contribs/database-plugin#144]
Hello,

I just re-installed new plugin version 4.9.3 after upgrading Archi to 4.9.1, but in the Preferences window, the Apply is always shows as gray, so I cannot save any database connection setting. Could you help?

Thanks a lot,
Xiaoqi
![TinyTake19-12-2021-10-36-18](https://user-images.githubusercontent.com/5661654/146681117-eea9050f-b78f-406c-ae05-c3fe9784b5a4.png)

**Comment by herve91** [comment]
Hi Xiaoqi,

That's strange. The button is not supposed to be greyed (I mean it should always be active, whatever a fonfiguration field has been updated or not).

Could you please provide me with mode information about your OS and version ? Are your using the JRE provided with Archi or another JRE ? How did you install the plugin (using the integrated Help / Manage plug ins menu or manually) ?

One more thing you may do to help me investigate: could you please activate the plugin log file:

Go to the preference page and click on the Logger tab. Then select the simple mode and the trace option. Hopefully, the Apply button will be activated on the Logger tab.

If the Apply button is still not activated, please go to the %APPDATA%\archi\.metadata\.plugins folder and edit the org.archicontribs.database.prefs file with a text editor. Please manually set the following options:
loggerFilename=C\:\\Users\\xiaoqi\\Documents\\DatabasePlugin.log
loggerLevel=trace
loggerMode=simple

Than close Archi, re-run it, open the database plugin preferences page, then close Archi again and provide me with the generated log file. I may find an arror message that provides me a clue about what could happen.

**Comment by steverumsby** [comment]
I had the same thing. Turning logging on, as you suggested above, fixed it, and it has been fine since. Before that I did uninstall and re-install the plugin, to no effect.

**Comment by herve91** [comment]
Hi Steve,
Following your post, I deducted that Xiaoqi issue was not isolated and therefore the issue was not coming from his desktop.

As I failed to reproduced it on my desktopn I instanciated a new VM and installed a brand new Archi and installed my plugin. And then, I experienced the same issue as Xiaoqi.

After investigation, I've found the issue. Part of release 4.9.3, I fixed an issue with the "browse" button on the log filename editor. Doing this, I did not realize that it would also activate a "isValid()" method that would provide the state of the log filename, and that the Eclipse preference page mechanism verifies all "isValid()" flags for all preference components.

The log filename answering "false" when it is not initialized (ie when the filename has not been set on the preference page, Eclipse deactivates the "Apply" button.

I fixed this behaviour and the fix will be in the 4.9.4 release.

Thanks for your message. It helped me finding where the issue was located.
